### PR TITLE
tremotesf: init at 2.4.0

### PR DESCRIPTION
**Pull request by sochotnicky**
-------

###### Description of changes

This adds a new Transmission daemon remote GUI: https://github.com/equeim/tremotesf2 

Tremotesf is written in Qt, and otherwise quite minimalistic.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 232488` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tremotesf</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -14689,6 +14689,12 @@
     githubId = 1437166;
     name = "Xia Bin";
   };
+  sochotnicky = {
+    email = "stanislav+github@ochotnicky.com";
+    github = "sochotnicky";
+    githubId = 55726;
+    name = "Stanislav Ochotnický";
+  };
```
Comment (Janik-Haag) [inline maintainers/maintainer-list.nix:14697]: This should be it's own commit before the tremotesf: init and called `maintainers: add sochotnicky`

File: pkgs/applications/networking/p2p/tremotesf/default.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+mkDerivation rec {
+  pname = "tremotesf";
+  version = "2.3.0";
+
+  doCheck = true;
+
+  src = fetchurl {
+    url = "https://github.com/equeim/tremotesf2/releases/download/${version}/${pname}-${version}.tar.gz";
+    sha256 = "sha256-/zRKPMdik4rDIoKL97zjJTfZ8iHmHTLXcDsWyjPBcJ8=";
+  };
```
Comment (Janik-Haag) [inline pkgs/applications/networking/p2p/tremotesf/default.nix:25]: You should use the [fetchFromGitHub](https://nixos.org/manual/nixpkgs/stable/#fetchfromgithub) wrapper here.

File: pkgs/applications/networking/p2p/tremotesf/default.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+, libpsl
+, cxxopts
+, wrapQtAppsHook
+}:
+
+mkDerivation rec {
+  pname = "tremotesf";
+  version = "2.3.0";
+
+  doCheck = true;
```
Comment (Janik-Haag) [inline pkgs/applications/networking/p2p/tremotesf/default.nix:20]: This is not needed because it's the implicit default

**Review by Janik-Haag — CHANGES_REQUESTED** [review]
Just a few nitpicks but overall quite a good PR :heart:

File: pkgs/applications/networking/p2p/tremotesf/default.nix (hunk 2 above)
Comment (sochotnicky) [reply pkgs/applications/networking/p2p/tremotesf/default.nix]: Done

File: pkgs/applications/networking/p2p/tremotesf/default.nix (hunk 3 above)
Comment (sochotnicky) [reply pkgs/applications/networking/p2p/tremotesf/default.nix]: I have removed it but interestingly now I don't see tests running when I build locally. I don't think I have changed anything but I suppose the CI/build system defaults are different than mine?

Comment (sochotnicky) [reply pkgs/applications/networking/p2p/tremotesf/default.nix]: > Just a few nitpicks but overall quite a good PR heart

Thank you for the review and the suggestions. I know this can be a thankless task

**Review by Janik-Haag — COMMENTED** [review]
Sorry I forgot that you want to add a desktopItem. For examples grep through nixpkgs. Will have to check if the tests still run, I was quite sure they would.

**Comment by sochotnicky** [comment]
> Sorry I forgot that you want to add a desktopItem. For examples grep through nixpkgs. Will have to check if the tests still run, I was quite sure they would.

Regarding the desktop item: Is that really needed? As far as I can tell cmake already installs desktop file provided in the package:
```
$ find /nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/32x32
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/32x32/apps
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/32x32/apps/org.equeim.Tremotesf.png
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/16x16
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/16x16/apps
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/16x16/apps/org.equeim.Tremotesf.png
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/256x256
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/256x256/apps
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/256x256/apps/org.equeim.Tremotesf.png
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/scalable
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/scalable/apps
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/scalable/apps/org.equeim.Tremotesf.svg
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/24x24
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/24x24/apps
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/24x24/apps/org.equeim.Tremotesf.png
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/48x48
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/48x48/apps
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/48x48/apps/org.equeim.Tremotesf.png
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/22x22
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/22x22/apps
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/icons/hicolor/22x22/apps/org.equeim.Tremotesf.png
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/applications
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/applications/org.equeim.Tremotesf.desktop
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/metainfo
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/share/metainfo/org.equeim.Tremotesf.appdata.xml
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/bin
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/bin/.tremotesf-wrapped
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/bin/tremotesf
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/nix-support
/nix/store/qyzba1brkjh4fcvqy724chl83f61qw32-tremotesf-2.3.0/nix-support/propagated-build-inputs
```

And it is available in my setup. But I concede that I could be missing some nuance/system integration piece.

**Comment by Janik-Haag** [comment]
You can add to your file and check:
```nix
packages = with pkgs; [
  (pkgs.callPackage ./tremotesf.nix { })
];
```
sorry for not responding.

**Comment by sochotnicky** [comment]
> You can add to your nix configuration and check:
> 
> ```nix
> packages = with pkgs; [
>   (pkgs.callPackage ./tremotesf.nix { })
> ];
> ```
> 
> sorry for not responding.

I do have the package installed and the desktop files show up properly in my environments. That's why I am confused.

**Comment by Janik-Haag** [comment]
I guess it's fine then.

**Review by Janik-Haag — APPROVED** [review]
LGTM

File: pkgs/applications/networking/p2p/tremotesf/default.nix [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+, qttools
+, kwidgetsaddons
+, kwindowsystem
+, fmt
+, libpsl
+, cxxopts
+, wrapQtAppsHook
+}:
+
+mkDerivation rec {
```
Comment (drupol) [inline pkgs/applications/networking/p2p/tremotesf/default.nix:16]: I think we should use the `finalAttrs` pattern here.

Since this is your first PR, let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

File: pkgs/applications/networking/p2p/tremotesf/default.nix [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  pname = "tremotesf";
+  version = "2.4.0";
+
+  src = fetchFromGitHub {
+    owner = "equeim";
+    repo = "tremotesf2";
+    rev = version;
+    sha256 = "sha256-TKtBgMpCWIUl1bohAKCbTcZX2uaPmzeWut/OeNs/rME=";
```
Comment (drupol) [inline pkgs/applications/networking/p2p/tremotesf/default.nix:24]:
```suggestion
    hash = "sha256-TKtBgMpCWIUl1bohAKCbTcZX2uaPmzeWut/OeNs/rME=";
```

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/p2p/tremotesf/default.nix (hunk 5 above)
Comment (Janik-Haag) [reply pkgs/applications/networking/p2p/tremotesf/default.nix:24]: sorry missed that :see_no_evil:

File: pkgs/applications/networking/p2p/tremotesf/default.nix (hunk 4 above)
Comment (sochotnicky) [reply pkgs/applications/networking/p2p/tremotesf/default.nix:16]: Thank you for the context/docs. Will tweak things to use finalAttrs & other fixes

**Review by drupol — APPROVED** [review]
Thanks for your contribution ! Keep them coming !
